src/test/java/seedu/address/logic/commands/AddCommandTest.java

```diff
@@ -157,6 +159,41 @@ public ObservableList<Person> getFilteredPersonList() {
         public void updateFilteredPersonList(Predicate<Person> predicate) {
             throw new AssertionError("This method should not be called.");
         }
+
+        @Override
+        public void setIncomeList(ReadOnlyIncomeList incomeList) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public ReadOnlyIncomeList getIncomeList() {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public boolean hasIncome(Income income) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void deleteIncome(Income target) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void addIncome(Income income) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public ObservableList<Income> getFilteredIncomeList() {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void updateFilteredIncomeList(Predicate<Income> predicate) {
+            throw new AssertionError("This method should not be called.");
+        }
```

Comment:
This feels like an anti-testing pattern, why are we creating test cases that just throw assertion errors? 🤔 Let's discuss cleaning this logic up, feels like this can be avoided